google-cloud-storage/src/test/java/com/google/cloud/storage/BucketTest.java

```diff
@@ -863,4 +863,51 @@ public void testDeleteLifecycleRules() {
     Bucket actualUpdatedBucket = updatedBucket.update();
     assertThat(actualUpdatedBucket.getLifecycleRules()).hasSize(0);
   }
+
+  @Test
+  public void testUpdateBucketLogging() {
+    initializeExpectedBucket(6);
+    BucketInfo.Logging logging =
+        BucketInfo.Logging.newBuilder()
+            .setLogBucket("logs-bucket")
+            .setLogObjectPrefix("test-logs")
+            .build();
+    BucketInfo bucketInfo =
+        BucketInfo.newBuilder("b")
+            .setAcl(ACLS)
+            .setEtag(ETAG)
+            .setGeneratedId(GENERATED_ID)
+            .setMetageneration(META_GENERATION)
+            .setOwner(OWNER)
+            .setSelfLink(SELF_LINK)
+            .setCors(CORS)
+            .setCreateTime(CREATE_TIME)
+            .setLogging(logging)
+            .setDefaultAcl(DEFAULT_ACL)
+            .setLifecycleRules(LIFECYCLE_RULES)
+            .setIndexPage(INDEX_PAGE)
+            .setNotFoundPage(NOT_FOUND_PAGE)
+            .setLocation(LOCATION)
+            .setStorageClass(STORAGE_CLASS)
+            .setVersioningEnabled(VERSIONING_ENABLED)
+            .setLabels(BUCKET_LABELS)
+            .setRequesterPays(REQUESTER_PAYS)
+            .setDefaultKmsKeyName(DEFAULT_KMS_KEY_NAME)
+            .setDefaultEventBasedHold(DEFAULT_EVENT_BASED_HOLD)
+            .setRetentionEffectiveTime(RETENTION_EFFECTIVE_TIME)
+            .setRetentionPeriod(RETENTION_PERIOD)
+            .setRetentionPolicyIsLocked(RETENTION_POLICY_IS_LOCKED)
+            .build();
```

Comment:
Nit: do you need all these settings here? Would it make sense to minimize invocation of setters on  the builder, just to make the code more compact.